improver/fire_weather/drought_code.py

```diff
@@ -0,0 +1,160 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+
+import numpy as np
+from iris.cube import Cube
+
+from improver.fire_weather import FireWeatherIndexBase
+
+
+class DroughtCode(FireWeatherIndexBase):
+    """
+    Plugin to calculate the Drought Code (DC) following
+    the Canadian Forest Fire Weather Index System.
+
+    The DC is a numerical rating of the average moisture content of deep,
+    compact organic layers. It is a useful indicator of seasonal drought
+    effects on forest fuels and the amount of smoldering in deep duff layers
+    and large logs. Higher values indicate drier conditions.
+
+    This process is adapted directly from:
+        Equations and FORTRAN Program for the
+        Canadian Forest Fire Weather Index System
+        (C.E. Van Wagner and T.L. Pickett, 1985).
+        Pages 6-7, Equations 18-23.
+
+    Expected input units:
+        - Temperature: degrees Celsius
+        - Precipitation: mm (24-hour accumulation)
+        - Previous DC: dimensionless
+        - Month: integer (1-12) for day length factor lookup
+    """
+
+    INPUT_CUBE_NAMES = [
+        "air_temperature",
+        "lwe_thickness_of_precipitation_amount",
+        "drought_code",
+    ]
+    OUTPUT_CUBE_NAME = "drought_code"
+    REQUIRES_MONTH = True
+    # Disambiguate input DC (yesterday's value) from output DC (today's calculated value)
+    INPUT_ATTRIBUTE_MAPPINGS = {"drought_code": "input_dc"}
+
+    temperature: Cube
+    precipitation: Cube
+    input_dc: Cube
+    month: int
+    previous_dc: np.ndarray
+
+    # Day length factors for DC calculation (L_f values from Table 4)
+    # Index 0 is unused, indices 1-12 correspond to months January-December
+    DC_DAY_LENGTH_FACTORS = [
+        0.0,  # Placeholder for index 0
+        -1.6,  # January
+        -1.6,  # February
+        -1.6,  # March
+        0.9,  # April
+        3.8,  # May
+        5.8,  # June
+        6.4,  # July
+        5.0,  # August
+        2.4,  # September
+        0.4,  # October
+        -1.6,  # November
+        -1.6,  # December
+    ]
+
+    def _calculate(self) -> np.ndarray:
+        """Calculate the Drought Code (DC).
+
+        Returns:
+            np.ndarray: The calculated DC values for the current day.
```

Comment:
```suggestion
            The calculated DC values for the current day.
```

Type not required in docstring.